internal/resources/cluster/integration/data_source_integration.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package integration
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster/integration"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+func DataSourceIntegration() *schema.Resource {
+	ri := &resourceIntegration{}
+
+	return &schema.Resource{
+		ReadContext: ri.dataSourceIntegrationRead,
+		Schema:      integrationSchema,
+	}
+}
+
+func (r *resourceIntegration) dataSourceIntegrationRead(_ context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	fn := constructFullName(d)
+
+	resp, err := r.client(m).ManageV1alpha1ClusterIntegrationResourceServiceRead(fn)
+	if err != nil {
+		if clienterrors.IsNotFoundError(err) {
+			d.SetId("")
+			return
+		}
+
+		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control integration entry, name: %s", fn.Name))
+	}
+
+	if resp == nil || resp.Integration == nil || resp.Integration.Meta == nil {
+		return diag.Errorf("invalid nil value reading resource %v", fn)
+	}
+
+	d.SetId(resp.Integration.Meta.UID)
+
+	// Meta
+	if err = d.Set(common.MetaKey, common.FlattenMeta(resp.Integration.Meta)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Status
+	// TODO
+
+	// FullName
+	if resp.Integration.FullName == nil {
+		return diag.Errorf("invalid nil full name reading resource %v", fn)
+	}
+
+	if err = d.Set(managementClusterNameKey, resp.Integration.FullName.ManagementClusterName); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err = d.Set(provisionerNameKey, resp.Integration.FullName.ProvisionerName); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err = d.Set(clusterNameKey, resp.Integration.FullName.ClusterName); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Spec
+	if resp.Integration.Spec == nil {
+		return diag.Errorf("invalid nil spec reading resource %v", fn)
+	}
+
+	switch resp.Integration.FullName.Name {
```

Comment:
If name can take only handful of inputs like TanzuServiceMesh and TanzuObservability. 
can you update the schema of this resource to have these checks.

The benefit we would get would be that, tf plan can perform validation and suggest these errors

Reply:
Yeah, I'll keep working on this - It's back in draft status again.